js/src/components/ads-account-select-control/index.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Internal dependencies
+ */
+import AppSelectControl from '.~/components/app-select-control';
+import useExistingGoogleAdsAccounts from '.~/hooks/useExistingGoogleAdsAccounts';
+
+/**
+ * @param {Object} props The component props
+ * @return {JSX.Element} An enhanced AppSelectControl component.
+ */
+const AdsAccountSelectControl = ( props ) => {
+	const { existingAccounts: accounts = [] } = useExistingGoogleAdsAccounts();
```

Comment:
In https://github.com/woocommerce/google-listings-and-ads/pull/2608#discussion_r1787274032:
> Before the accounts are fetched, the returned `existingAccounts` is `null` and won't fall back to `[]`, [...]

The ineffective fallback can be removed to minimize unnecessary logic branches.

Reply:
👍